services/property-svc/internal/property/projections/property_postgres_projection/property_postgres_projection.go

```diff
@@ -0,0 +1,614 @@
+package property_postgres_projection
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	pb "gen/proto/services/property_svc/v1"
+	"github.com/EventStore/EventStore-Client-Go/v4/esdb"
+	"github.com/google/uuid"
+	"github.com/jackc/pgx/v5"
+	zlog "github.com/rs/zerolog/log"
+	"hwdb"
+	"hwes"
+	"hwes/eventstoredb/projections/custom"
+	"hwutil"
+	propertyEventsV1 "property-svc/internal/property/events/v1"
+	"property-svc/internal/property/models"
+	"property-svc/repos/property_repo"
+)
+
+type Projection struct {
+	*custom.CustomProjection
+	propertyRepo *property_repo.Queries
+}
+
+func NewProjection(es *esdb.Client, serviceName string) *Projection {
+	subscriptionGroupName := fmt.Sprintf("%s-postgres-projection", serviceName)
+	p := &Projection{custom.NewCustomProjection(es, subscriptionGroupName), property_repo.New(hwdb.GetDB())}
+	p.initEventListeners()
+	return p
+}
+
+func (p *Projection) initEventListeners() {
+	p.RegisterEventListener(propertyEventsV1.PropertyCreated, p.onPropertyCreated)
+	p.RegisterEventListener(propertyEventsV1.PropertyDescriptionUpdated, p.onPropertyDescriptionUpdated)
+	p.RegisterEventListener(propertyEventsV1.PropertySetIDUpdated, p.onPropertySetIDUpdated)
+	p.RegisterEventListener(propertyEventsV1.PropertySubjectTypeUpdated, p.onSubjectTypeUpdated)
+	p.RegisterEventListener(propertyEventsV1.PropertyFieldTypeUpdated, p.onFieldTypeUpdated)
+	p.RegisterEventListener(propertyEventsV1.PropertyNameUpdated, p.onNameUpdated)
+	p.RegisterEventListener(propertyEventsV1.PropertyArchived, p.onPropertyArchived)
+	p.RegisterEventListener(propertyEventsV1.PropertyRetrieved, p.onPropertyRetrieved)
+	p.RegisterEventListener(propertyEventsV1.PropertyFieldTypeDataCreated, p.onPropertyFieldTypeDataCreated)
+	p.RegisterEventListener(propertyEventsV1.PropertyFieldTypeDataAllowFreetextUpdated, p.onAllowFreetextUpdated)
+	p.RegisterEventListener(propertyEventsV1.PropertyFieldTypeDataSelectOptionsRemoved, p.onFieldTypeDataSelectOptionsRemoved)
+	p.RegisterEventListener(propertyEventsV1.PropertyFieldTypeDataSelectOptionsUpserted, p.onFieldTypeDataSelectOptionsUpserted)
+}
+
+func (p *Projection) onPropertyCreated(ctx context.Context, evt hwes.Event) (error, esdb.NackAction) {
+	log := zlog.Ctx(ctx)
+	db := hwdb.GetDB()
+
+	tx, err := db.Begin(ctx)
+	if err != nil {
+		return err, esdb.NackActionRetry
+	}
+
+	defer func() {
+		err := tx.Rollback(ctx)
+		if !errors.Is(err, pgx.ErrTxClosed) {
+			log.Error().Err(err).Msg("rollback failed.")
+		}
+	}()
+
+	propertyRepo := property_repo.New(db).WithTx(tx)
+
+	var payload propertyEventsV1.PropertyCreatedEvent
+	if err := evt.GetJsonData(&payload); err != nil {
+		log.Error().Err(err).Msg("unmarshal failed")
```

Comment:
payload might be interesting

Reply:
I would say the error is enough here... Ofc, I can add it if you'd like. Your decison.